modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -14,38 +14,51 @@
  *
  */
 public class KafkaContainer extends GenericContainer<KafkaContainer> {
+    public static final int DEFAULT_KAFKA_PORT = 9093;
 
-    public static final int KAFKA_PORT = 9093;
+    public static final String DEFAULT_KAFKA_BROKER_ID = "1";
+
+    public static final int DEFAULT_INTERNAL_TOPIC_RF = 1;
 
     public static final int ZOOKEEPER_PORT = 2181;
 
+    public static final String CONFLUENT_PLATFORM_VERSION = "5.2.1";
+
     private static final int PORT_NOT_ASSIGNED = -1;
 
     protected String externalZookeeperConnect = null;
 
     private int port = PORT_NOT_ASSIGNED;
+    private int exposedPort;
 
     public KafkaContainer() {
-        this("5.2.1");
+        this(CONFLUENT_PLATFORM_VERSION);
     }
 
     public KafkaContainer(String confluentPlatformVersion) {
+        this(confluentPlatformVersion, DEFAULT_KAFKA_PORT, DEFAULT_KAFKA_BROKER_ID, DEFAULT_INTERNAL_TOPIC_RF);
+    }
+
+    public KafkaContainer(String confluentPlatformVersion, int exposedPort, String brokerId, int internalTopicRf) {
         super(TestcontainersConfiguration.getInstance().getKafkaImage() + ":" + confluentPlatformVersion);
 
+        this.exposedPort = exposedPort;
         // TODO Only for backward compatibility
         withNetwork(Network.newNetwork());
         withNetworkAliases("kafka-" + Base58.randomString(6));
-        withExposedPorts(KAFKA_PORT);
+        withExposedPorts(exposedPort);
```

Comment:
I don't think we need to override the internal port. Each container will have its own host, so that they can use the same port, and the tests will see a random port anyways